app/code/Bulbulatory/Recommendations/Model/Total/Quote/RecommendationDiscount.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+
+namespace Bulbulatory\Recommendations\Model\Total\Quote;
+
+
+use Bulbulatory\Recommendations\Helper\Config;
+use Bulbulatory\Recommendations\Helper\Recommendation as RecommendationHelper;
+use Magento\Framework\Phrase;
+use Magento\Quote\Api\Data\ShippingAssignmentInterface;
+use Magento\Quote\Model\Quote;
+use Magento\Quote\Model\Quote\Address\Total;
+use Magento\Quote\Model\Quote\Address\Total\AbstractTotal;
+use Magento\Customer\Model\Session;
+
+/**
+ * Class RecommendationDiscount
+ * @package Bulbulatory\Recommendations\Model\Total\Quote
+ */
+class RecommendationDiscount extends AbstractTotal
+{
+    /**
+     * @var int
+     */
+    private $recommendationDiscount = 0;
+    /**
+     * @var Phrase
+     */
+    private $discountLabel = '';
+
+    /**
+     * RecommendationDiscount constructor.
+     * @param Session $session
+     * @param RecommendationHelper $recommendationHelper
+     * @param Config $config
+     */
+    public function __construct(Session $session, RecommendationHelper $recommendationHelper, Config $config)
+    {
+        if ($config->isRecommendationsModuleEnabled()) {
+            $confirmedRecommendations = $recommendationHelper->getConfirmedRecommendationsCountForCustomer($session->getCustomer());
+            $this->recommendationDiscount = $recommendationHelper->getDiscountValueForCustomer($confirmedRecommendations);
+            $this->discountLabel = __('Recommendations discount -%1%', $this->recommendationDiscount);
+        }
+    }
+
+    /**
+     * @param Quote $quote
+     * @param ShippingAssignmentInterface $shippingAssignment
+     * @param Total $total
+     * @return $this|RecommendationDiscount
+     */
+    public function collect(
+        Quote $quote,
+        ShippingAssignmentInterface $shippingAssignment,
+        Total $total
+    )
+    {
+        parent::collect($quote, $shippingAssignment, $total);
+        $items = $shippingAssignment->getItems();
+        if (!count($items)) {
+            return $this;
+        }
+
+        if ($this->recommendationDiscount <= 0) {
+            return $this;
+        }
```

Comment:
Z wydajnościowego punktu widzenia lepiej te dwa bloki zamienić miejscami - sprawdzenie wartości zniżki, która i tak została ustawiona w konstruktorze, to chwila. Wykonanie funkcji na modelu i potem jeszcze `count` na otrzymanej wartości jest bardziej czasochłonny, a nie ma sensu tego robić jeśli i tak zniżki naliczać nie będziemy.